src/libexpr/eval-settings.hh

```diff
@@ -19,11 +19,26 @@ struct EvalSettings : Config
     Setting<Strings> nixPath{
         this, getDefaultNixPath(), "nix-path",
         R"(
-          List of directories to be searched for `<...>` file references
-
-          In particular, outside of [pure evaluation mode](#conf-pure-eval), this determines the value of
+          List of search paths to use for lookup paths.
```

Comment:
```suggestion
          List of search paths to use for lookup paths and `builtins.nixPath` for impure eval.
          With pure eval it is ignored entirely.
```
I would mention pure vs impure eval right at the top. This option is dead code with pure eval and that's important to know.